mne/report.py

```diff
@@ -965,6 +967,100 @@ def add_bem_to_section(self, subject, caption='BEM', section='bem',
         self._sectionlabels.append(sectionvar)
         self.html.extend(html)
 
+    def add_slider_to_section(self, figs, captions=None, section='custom',
+                              scale=None, image_format='png'):
+        """Renders a slider of figs to the report.
+
+        Parameters
+        ----------
+        figs : list of figures.
+            Each figure in the list can be an instance of
+            matplotlib.pyplot.Figure, mayavi.core.scene.Scene,
+            or np.ndarray (images read in using scipy.imread).
+        captions : list of str | float | None
+            A list of captions to the figures. If float, a str will be
+            constructed as `%f s`. If None, it will default to
+            `Data slice %d`.
+        section : str
+            Name of the section. If section already exists, the figures
+            will be appended to the end of the section
+        scale : float | None | callable
+            Scale the images maintaining the aspect ratio.
+            If None, no scaling is applied. If float, scale will determine
+            the relative scaling (might not work for scale <= 1 depending on
+            font sizes). If function, should take a figure object as input
+            parameter. Defaults to None.
+        image_format : {'png', 'svg'}
+            The image format to be used for the report. Defaults to 'png'.
+
+        Notes
+        -----
+        .. versionadded:: 0.10.0
+        """
+
+        _check_scale(scale)
+        if not isinstance(figs[0], list):
+            figs = [figs]
+        else:
+            raise NotImplementedError('`add_slider_to_section` '
+                                      'can only add one slider at a time.')
+        figs, _, _ = self._validate_input(figs, section, section)
+
+        sectionvar = self._sectionvars[section]
+        global_id = self._get_id()
+        div_klass = self._sectionvars[section]
+        img_klass = self._sectionvars[section]
+        name = 'slider'
+
+        html = []
+        html.append(u'<li class="slider" id="%d">\n' % global_id)
+        html.append(u'<h2>%s</h2>\n' % section)
+        html.append(u'<div class="row">')
+
+        html.append(u'<div class="col-xs-6 col-md-4">')
+        slides_klass = '%s-%s' % (name, global_id)
+
+        if isinstance(figs[0], list):
+            figs = figs[0]
+        sl = np.arange(0, len(figs))
+        slices = []
+        img_klass = 'slideimg-%s' % name
+        div_klass = 'span12 %s' % slides_klass
+
+        if isinstance(captions[0], float):
+            captions = ['%0.3f s' % caption for caption in captions]
```

Comment:
I think you should add a test for the caption length.

If you have `_add_slider_to_section([fig] * 10, 'abcd')`, it will stop displaying the figure after the 4th one.